kuka_controllers/src/joint_impedance_controller.cpp

```diff
@@ -55,14 +69,52 @@ const
 controller_interface::CallbackReturn
 JointImpedanceController::on_configure(const rclcpp_lifecycle::State &)
 {
+  // Check if ParaListner is working correctly
+  if (!param_listener_)
+  {
+    RCLCPP_ERROR(get_node()->get_logger(), "Error encountered during init");
+    return controller_interface::CallbackReturn::ERROR;
+  }
+
+  // update the dynamic map parameters
+  param_listener_->refresh_dynamic_parameters();
+
+  // get parameters from the listener in case they were updated
+  params_ = param_listener_->get_params();
+
+  // Check if the DoF has changed
+  if ((dof_ > 0) && (params_.joints.size() != dof_))
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(),
+      "The JointImpedanceController does not support restarting with a different number of DOF");
+    return CallbackReturn::FAILURE;
+  }
+
+  if (params_.joints.empty())
```

Comment:
is it ok if configuration is successful in this case? We should maybe discuss this

Reply:
ok